server/src/controllers/GrailController.ts

```diff
@@ -86,8 +102,157 @@ export class GrailController {
       dataToSet.runewordData = runewordGrailData;
       return dataToSet;
     });
+
+    this.getMemberParties(address, parties => {
+      parties.forEach(party => {
+        this.saveGrailInPartyDatabase(address, party.address, grailData);
+      });
+    });
   };
 
+  public saveGrailInPartyDatabase = async (
+    grailAddress: string,
+    partyAddress: string,
+    data: any
+  ) => {
+    let partyUserData = GrailController.formatGrailForParty(data);
+    const result = await this.partyCollection.findOneAndUpdate(
+      {
+        address: GrailController.trimAndToLower(partyAddress),
+        "data.users.username": grailAddress
+      },
+      {
+        $set: {
+          "data.users.$.data": partyUserData
+        },
+        $inc: { updateCount: 1 }
+      },
+      { returnOriginal: false }
+    );
+  };
+
+  public static formatGrailForParty = (data: any): any => {
+    let partyGrailData = {
+      uniqueArmor: {
+        missing: 123
+      },
+      uniqueWeapons: {
+        missing: 197
+      },
+      uniqueOther: {
+        missing: 59
+      },
+      sets: {
+        missing: 127
+      },
+      itemScore: 0
+    };
+    if (data && data.uniques) {
+      if (data.uniques.weapons) {
+        let missingWeps = GrailController.sumMissing(
+          () => data.uniques.weapons,
+          new MissingItems()
+        );
+        partyGrailData.uniqueWeapons.missing = missingWeps.missing;
```

Comment:
Note to self: This is the only really critical file, so I will review this again, once you've done the changes (if you agree of course :D )